python/dgl/graphbolt/impl/sampled_subgraph_impl.py

```diff
@@ -68,6 +68,63 @@ def __post_init__(self):
             ), "Nodes in pairs should be of type torch.Tensor."
 
 
+def _to_reverse_ids(node_pair, reverse_row_node_ids, reverse_column_node_ids):
+    u, v = node_pair
+    if reverse_row_node_ids is not None:
+        u = reverse_row_node_ids[u]
+    if reverse_column_node_ids is not None:
+        v = reverse_column_node_ids[v]
+    return (u, v)
+
+
+def _relabel_homo_edges(lhs_edges, rhs_edges):
+    """Relabel the source nodes and destination nodes independently for two set
+    of edges."""
+    src = torch.cat([lhs_edges[0], rhs_edges[0]])
+    dst = torch.cat([lhs_edges[1], rhs_edges[1]])
+    _, src = torch.unique(src, return_inverse=True)
+    _, dst = torch.unique(dst, return_inverse=True)
+    lhs_edges = (src[: lhs_edges[0].numel()], dst[: lhs_edges[0].numel()])
+    rhs_edges = (src[lhs_edges[0].numel() :], dst[lhs_edges[0].numel() :])
+    return lhs_edges, rhs_edges
+
+
+def _exclude_homo_edges(lhs_edges, rhs_edges):
+    lhs_edges, rhs_edges = _relabel_homo_edges(lhs_edges, rhs_edges)
+    # Compact edges to integers to use torch.isin to exclude edges
+    dst_range = lhs_edges[0].numel() + rhs_edges[0].numel()
+    lhs_val = lhs_edges[0] * dst_range + lhs_edges[1]
```

Comment:
Can I understand first element in tuple as high number and the second as low number with  `dst_range` as radix?
If this is the case, will it cause error because both` lhs_edges[1]` and `rhs_edges[1]` can be larger than `dst_range`?

Reply:
They would not be larger than `lhs_edges[0].numel() + rhs_edges[0].numel()` since they are compacted.